rj_gameplay/rj_gameplay/tactic/goalie_tactic.py

```diff
@@ -0,0 +1,134 @@
+    wall_vec = np.array([dir_vec[1], -dir_vec[0]])
+
+    # find and return pt
+    mid_pt = goal_pt + (dir_vec * PCT_TO_BALL * np.linalg.norm(ball_pt - goal_pt))
+    return mid_pt
+
+
+class GoalieTactic(tactic.ITactic):
+    def __init__(self):
+
+        # create move SkillEntry
+        self.move_se = tactic.SkillEntry(move.Move())
+
+        self.receive_se = tactic.SkillEntry(receive.Receive())
+
+        # TODO: rename cost_list to role_cost in other gameplay files
+        self.role_cost = goalie_cost()
+
+    def compute_props(self):
+        pass
+
+    def create_request(self, **kwargs) -> role.RoleRequest:
+        """Creates a sane default RoleRequest.
+        :return: A list of size 1 of a sane default RoleRequest.
+        """
+        pass
+
+    def get_requests(self, world_state: rc.WorldState,
+                     props) -> List[tactic.RoleRequests]:
+        """
+        :return: A list of role requests for move skills needed
+        """
+
+        # TODO: this const is copy-pasted from wall_tactic 
+        # put into common param file: https://www.geeksforgeeks.org/global-keyword-in-python/
+
+        # dist is slightly greater than penalty box bounds
+        box_w = world_state.field.penalty_long_dist_m
+        box_h = world_state.field.penalty_short_dist_m
+        line_w = world_state.field.line_width_m
+        MIN_WALL_RAD = RobotConstants.RADIUS + line_w + np.hypot(box_w / 2, box_h)
+
+        role_requests = {}
+        if world_state and world_state.ball.visible:
+            ball_to_goal_dist = np.linalg.norm(world_state.field.our_goal_loc - world_state.ball.pos)
+            if ball_to_goal_dist < MIN_WALL_RAD:
+                # move to block when ball inside wall
+
+                # create RoleRequest for each SkillEntry
+                role_requests[self.receive_se] = [role.RoleRequest(role.Priority.HIGH, False, self.role_cost)]
+            else:
+                # else, track ball normally
+
+                # update move skill
+                self.move_se.skill.target_point = get_goalie_pt(world_state)
+                self.move_se.skill.face_point = world_state.ball.pos
+
+                # create RoleRequest for each SkillEntry
+                role_requests[self.move_se] = [role.RoleRequest(role.Priority.HIGH, False, self.role_cost)]
+        return role_requests
+
+    def tick(self,
+             role_results: tactic.RoleResults) -> List[tactic.SkillEntry]:
+        """
+        :return: A list of skills depending on which roles are filled
+        """
+
+        # create list of skills based on if RoleResult exists for SkillEntry
+        skills = []
+
+        move_result = role_results[self.move_se]
+        if move_result and move_result[0].is_filled():
+            skills.append(self.move_se)
+
+        receive_result = role_results[self.receive_se]
+        if receive_result and receive_result[0].is_filled():
+            skills.append(self.receive_se)
```

Comment:
This should probably be an if else to match the logic from get_requests (also on the off chance both receive and move get assigned)